actix-router/src/resource.rs

```diff
@@ -196,9 +403,37 @@ impl ResourceDef {
         }
     }
 
-    /// Is prefix path a match against this resource.
-    pub fn is_prefix_match(&self, path: &str) -> Option<usize> {
-        profile_method!(is_prefix_match);
+    /// Tries to match prefix of `path` to this resource, returning the position in the path where
+    /// the prefix match ends.
+    ///
+    /// # Examples
+    /// ```
+    /// use actix_router::ResourceDef;
+    ///
+    /// // static resource does not do prefix matching
+    /// let resource = ResourceDef::new("/user");
+    /// assert_eq!(resource.find_match("/user"), Some(5));
+    /// assert!(resource.find_match("/user/").is_none());
+    /// assert!(resource.find_match("/user/123").is_none());
+    /// assert!(resource.find_match("/foo").is_none());
+    ///
+    /// // constant prefix resource
+    /// let resource = ResourceDef::prefix("/user");
+    /// assert_eq!(resource.find_match("/user"), Some(5));
+    /// assert_eq!(resource.find_match("/user/"), Some(5));
+    /// assert_eq!(resource.find_match("/user/123"), Some(5));
+    /// assert!(resource.find_match("/foo").is_none());
+    ///
+    /// // dynamic prefix resource
+    /// let resource = ResourceDef::prefix("/user/{id}");
+    /// assert_eq!(resource.find_match("/user/123"), Some(9));
+    /// assert_eq!(resource.find_match("/user/123/"), Some(9));
+    /// assert_eq!(resource.find_match("/user/123/stars"), Some(9));
+    /// assert!(resource.find_match("/user/").is_none());
+    /// assert!(resource.find_match("/foo").is_none());
+    /// ```
+    pub fn find_match(&self, path: &str) -> Option<usize> {
+        profile_method!(find_match);
 
         let path_len = path.len();
         let path = if path.is_empty() { "/" } else { path };
```

Comment:
```suggestion
```
This should be removed to have a consistent behavior with `is_match` and `match_path`. This also fixes the bug with empty patterns I mentioned earlier.